packages/ra-ui-materialui/src/detail/showFieldTypes.tsx

```diff
@@ -81,14 +83,16 @@ ${children.map(child => `            ${child.getRepresentation()}`).join('\n')}
     referenceArray: {
         component: ReferenceArrayField,
         representation: (props: ReferenceArrayFieldProps) =>
-            `<ReferenceArrayField source="${props.source}" reference="${props.reference}"><TextField source="id" /></ReferenceArrayField>`,
+            `<ReferenceArrayField source="${props.source}" reference="${props.reference}" />`,
     },
     referenceArrayChild: {
-        component: (
-            props: { children: ReactNode } & Omit<InputProps, 'source'> &
-                Partial<Pick<InputProps, 'source'>>
-        ) => <TextField source="id" {...props} />,
-        representation: () => `<TextField source="id" />`,
+        component: () => (
+            <SingleFieldList>
+                <ChipField source="id" />
+            </SingleFieldList>
+        ),
+        representation: () =>
+            `<SingleFieldList><ChipField source="id" /></SingleFieldList>`,
```

Comment:
Can't you remove the `referenceArrayChild` key altogether? I'm worried the rendered JSX does not match the JSX code printed in the console.

Reply:
It might be used by guessers we don't maintain so I'd rather keep it